File: src/Totem.Application/Services/PasswordServices/IPasswordService.cs
using Totem.Common.Services;
using Totem.Domain.Models.PasswordModels;

namespace Totem.Application.Services.PasswordServices
{
	public interface IPasswordService
	{
		Task<(Result result, PasswordView data)> GetByIdPasswordAsync(Guid id);
		Task<(Result result, List<PasswordView> data)> GetListPasswordAsync();
		Task<(Result result, Guid data)> AddPasswordAsync(PasswordRequest request);
		Task<Result> RemovePasswordAsync(Guid id);
		Task<(Result result, PasswordView data)> AssignNextPasswordAsync(Guid queueId, Guid serviceLocationId, string newServiceLocationName);
		Task<Result> TransferPasswordAsync(Guid passwordId, PasswordTransferRequest passwordTransfer);
		Task<Result> MarkAsServed(Guid passwordId);

	}
}

File: src/Web/src/services/interfaces/IPasswordService.ts
import type { ServiceResult } from "../../models/baseServiceModels";
import type { PasswordView, PasswordRequest, PasswordTransferRequest } from "../../models/PasswordModels";

export interface IPasswordService {
    getPasswordByIdAsync(id: string): Promise<ServiceResult<PasswordView>>;
    getPasswordsAsync(queueId: string): Promise<ServiceResult<PasswordView[]>>;
    addPasswordAsync(request: PasswordRequest): Promise<ServiceResult<string>>;
    transferPasswordAsync(passwordId: string, request: PasswordTransferRequest): Promise<ServiceResult<void>>;
    markAsServedAsync(passwordId: string): Promise<ServiceResult<void>>;
    removePasswordAsync(id: string): Promise<ServiceResult<void>>;
}
